packages/esm-patient-common-lib/src/workspaces/useWorkspaces.tsx

```diff
@@ -30,10 +33,24 @@ export function useWorkspaces(): WorkspacesInfo {
     }
   }, [workspaces]);
 
-  return {
-    active: workspaces.length > 0,
-    windowState,
-    workspaces,
-    prompt,
-  };
+  useEffect(() => {
+    if (workspaces.length === 0) {
+      updateWorkspaceSizeState('hidden');
+    } else {
+      updateWorkspaceSizeState(workspaces[0].preferredWindowSize === 'maximized' ? 'maximized' : 'normal');
+    }
+  }, [workspaces]);
+
+  const memoisedResults = useMemo(
```

Comment:
I actually don't know that we need memoisation here. There's nothing very expensive about constructing an object.

Reply:
No, it isn't expensive, but what usually happens that the returned object gets created everytime on re-render of the hook. Hence the reference of the returned object changes on re-render. The values of the object are used as dependency in other hooks and hence can trigger corresponding hooks.

But now that I think, the inner values are memoised, hence the values' reference will not change 🤔 .

Reply:
Hard to know with these things. I think the usual guidance is to use `useMemo` when A) you are quite sure that not using it will cause a problem or B) after you've noticed a performance problem that would be fixed by `useMemo`.

In this case, since we always use `useWorkspaces` with destructuring, it doesn't matter whether the return object itself changes.